### PR TITLE
Properly generate a method off of a property in VB.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/11461

This was failing before because of the unending tail of problems around trying to map from a compiler error to the appropriate node in question.  We would previously look from the most narrowest node to the widest node possible.  This was a problem because a narrower node might look like something we could generate a method for.  

We now look for this and we don' try generating methods on prefix expressions of the node we have the error on.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
test vsi please

**Comment by CyrusNajmabadi** [comment]
retest vsi please

**Comment by CyrusNajmabadi** [comment]
tagging @dotnet/roslyn-ide

File: src/EditorFeatures/VisualBasicTest/Diagnostics/GenerateMethod/GenerateMethodTests.vb [hunk 1]
```diff
@@ -2232,6 +2232,52 @@ End Namespace
...
+    Friend Sub NewFunction(v1 As String, v2 As String)
+        Throw New NotImplementedException()
+    End Sub
+End Class
+
+Class C
+    Shared Sub TestError()
+        Repository.AgreementType.NewFunction("", "")
+    End Sub
+End Class</text>.Value.Replace(vbLf, vbCrLf))
```
Comment (jasonmalinowski) [inline src/EditorFeatures/VisualBasicTest/Diagnostics/GenerateMethod/GenerateMethodTests.vb]: We do have .NormalizedValue for this.

**Comment by jasonmalinowski** [comment]
:+1:

File: src/EditorFeatures/VisualBasicTest/Diagnostics/GenerateMethod/GenerateMethodTests.vb (hunk 1 above)
Comment (DustinCampbell) [inline src/EditorFeatures/VisualBasicTest/Diagnostics/GenerateMethod/GenerateMethodTests.vb]: Or, just use an actual multiline string. :smile:

**Pull request merged**